src/FixedPointNumbers.jl

```diff
@@ -42,25 +40,202 @@ export
     # Functions
     scaledual
 
+# T => BaseType
+# f => Number of Bytes reserved for fractional part
+immutable FixedPoint{T <: Integer, f} <: Real
+    i::T
+
+    # constructor for manipulating the representation;
+    # selected by passing an extra dummy argument
+    FixedPoint(i::T, _) = new(i)
+    FixedPoint(i::Integer,_) = new(i % T)
```

Comment:
`i % T` is a new change and maybe it is better to throw a DomainError, at least we should add a test for this